relay-server/src/utils/rate_limits.rs

```diff
@@ -102,6 +102,13 @@ fn infer_event_category(item: &Item) -> Option<DataCategory> {
         ItemType::MetricBuckets => None,
         ItemType::FormData => None,
         ItemType::UserReport => None,
+        // the following items are "internal" item types.  From the perspective of the SDK
+        // the use the "internal" data category however this data category is in fact never
+        // supposed to be emitted by relay as internal items must not be rate limited.  As
+        // such we do not emit a data category here.  An SDK however is supposed to assume
+        // that such item types use the reserved "internal" data category to not accidentally
+        // assume another rate limit (such as default).
```

Comment:
same comment as before, why have this datacategory if it is never used by relay. The fact that it's called internal seems like an impl detail not relevant to relay, clients should just never let rate limits affect that item, right?

Would add mention of "internal" data category to docstring of the enum if you want to define that in Relay, but I don't understand how Relay needs to know about this.

Reply:
Closing this comment as resolved, keeping the discussion above.